backend/inspirehep/submissions/views.py

```diff
@@ -160,21 +157,32 @@ def get_updated_record_data(self, update_data, record):
             "arxiv_categories",
             "advisors",
         ]
-        try:
-            updated_ids = update_data["ids"]
-            updated_schemas = get_value(updated_ids, "schema", default=[])
-            updated_orcid = get_values_for_schema(updated_ids, "ORCID")
-            record_ids = record.pop("ids", [])
-            for record_id in record_ids:
-                if (record_id["schema"] not in updated_schemas) or (
-                    record_id["schema"] == "ORCID"
-                    and record_id["value"] != updated_orcid[0]
-                ):
-                    update_data["ids"].append(record_id)
-        except KeyError:
-            pass
 
-        return get_updated_record_data(record, update_data, optional_fields)
+        for field in optional_fields:
+            if field not in set(update_data) and field in set(dict(record)):
+                del record[field]
+
+        if "_private_notes" in update_data and "_private_notes" in record:
+            record["_private_notes"].extend(update_data["_private_notes"])
+            del update_data["_private_notes"]
+
+        if "ids" in update_data:
+            record_ids = record.get("ids", [])
+            record_ids_schemas = set(get_value(record, "ids.schema", []))
+            for id_dict in update_data["ids"]:
+                if (id_dict["schema"] not in record_ids_schemas) or (
+                    id_dict["schema"] == "ORCID" and id_dict not in record_ids
```

Comment:
Shouldn't this be like the following?

```suggestion
                    id_dict["schema"] == "ORCID" and id_dict['value'] not in record_ids
```

Reply:
if it's true then we need to add tests for this case too.

Reply:
I think that there is no difference if it will be `id_dict['value']` or `id_dict` because the schema in this case should always be `ORCID`.

We're already testing update when orcid is not in the record ids here https://github.com/inspirehep/inspirehep/blob/17a0dec6bbd9d9a8bd27b94e99afa278dde58250/backend/tests/integration/submissions/test_submissions_views.py#L312
 and in other tests we're testing the case when other ids from update data are missing in record ids for example linkedin here
https://github.com/inspirehep/inspirehep/blob/6bdf7ce1f9a0b8b977e5f13c1aa735119db387d6/backend/tests/integration/submissions/test_submissions_views.py#L283